nucypher/config/util.py

```diff
@@ -0,0 +1,33 @@
+"""
+ This file is part of nucypher.
+
+ nucypher is free software: you can redistribute it and/or modify
+ it under the terms of the GNU Affero General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ nucypher is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU Affero General Public License for more details.
+
+ You should have received a copy of the GNU Affero General Public License
+ along with nucypher.  If not, see <https://www.gnu.org/licenses/>.
+"""
+
+from pathlib import Path
+from typing import Optional, get_type_hints
+
+
+def cast_paths_from(cls, payload):
```

Comment:
Some docstring would be useful. 

Relying on typehints being present and correct is kind of brittle. Personally, I'd go for
- corresponding classes accepting either a `Path` or a `str` and casting it; or
- using `marshmallow` to deserialized from JSON

Reply:
I would prefer the latter, as IMHO we would benefit from having the serialization format described in `marshmallow`. I can address this in a different PR.